### ISSUE TITLE
DeleteFromScope

### ISSUE THREAD

**Issue by da-x** [cling-lang/spec#2]
The idea is to remove local variables from scope without using enclosing curly braces. This is useful when you want to prevent access to these variables by accident due to offhanded code edits. The name can or cannot be reused by new declaration, depending on how it was removed.

```c
%pragma cling DeleteFromScope

int x = 4;
int y = 3;
printf("%d %d\n", x, y); // Works

% x; // Remove x, don't allow reuse
%% y; // Remove x, allow reuse

x = 2;
printf("x\n", 1); // Compilation error - x is no longer in scope.
char x; // does not compile.
char y; // compiles
```
Here we use `% [list of identifiers]` as a statement a valid statement, to avoid introducing new keywords.

**Comment by da-x** [comment]
@sinelaw

**Comment by sinelaw** [comment]
No-reuse can be 'tier 1' if we can find a C11-valid syntax for it (maybe something involving an attribute?)

In other words, it's backwards compatible. So let's make the syntax also backwards compatible.

Maybe this:
```C

int x;
// bla bla bla
__delete_var(x);

x += 2; // warning
```

Where '__delete_var' is a #define:
```C
#ifdef FEATURE_DELETE_NO_REUSE
#define __delete_var(x) __attribute__((delete_no_reuse(x)))
#else
#define __delete_var(x)
#endif 
```

**Comment by sinelaw** [comment]
P.S. Perhaps `done_using` is a better name

**Comment by da-x** [comment]
Okay, so we would need a 'standalone attribute-statement' concept. That might be useful for other extensions as well.

**Comment by sinelaw** [comment]
Clang has this for fallthrough
